### PR TITLE
work on docs

### PR DESCRIPTION
**Pull request by nadavwix**

### CODE REVIEW

File: docs/syntax.md [hunk 1]
```diff
@@ -60,17 +60,15 @@ const  = TSXAir(() => {
...
-- Changes to a [when](#using-when)/[memo](#using-memo) dependency
 - Calling `invalidate`
 
 ## "when" and "memo"
-`when` and `memo` define actions that are re-evaluated when their dependencies change.
-The dependencies can be defined or, if not specified inferred from the action code.
+`when` and `memo` allow running parts of the component logic only when a specific dependency has changed.
 ### Diffrences between when and memo:
-- a `when` action may return an "undo" function that will be called before the action is called again
+- a `when` action may return an "undo" function that will be called before the action is called again ( due to dependency change ) and when the component is about to be unmounted
```
Comment (omryn) [inline docs/syntax.md:71]: I think we only make a call to the **undo** function before unmounting (not the main action). I don't think an unmount is a natural change to explicit dependencies

File: docs/syntax.md [hunk 2]
```diff
@@ -139,7 +137,10 @@ const InfiniteMeasure = TSXAir(() => {
 ```
 
 ### beforeUnmount
+
 ```tsx
+// @omry, this is an example of why you want to return a dispose function from afterMount
```
Comment (omryn) [inline docs/syntax.md:142]: Agreed. Can you think of a use case for  just beforeUnmount?

File: docs/syntax.md [hunk 3]
```diff
@@ -126,20 +124,25 @@ const GoogleMaps = TSXAir(()=>{
...
 
-### afterUpdate
+### afterDomUpdate
 ```tsx
 const InfiniteMeasure = TSXAir(() => {
     const state = store({area:0, ref:{} as RefHolder<HTMLDivElement>});   
-    afterDomUpdate([], ()=>{
-        const {width, height} = state.ref.element!.getClientRects()[0];
-        state.area = width * height;
+    afterDomUpdate([], (consecutiveUpdatedFrames)=>{
```
Comment (omryn) [inline docs/syntax.md:131]: @nadavwix what do you think of this as a way to limit infinite loops potential?

File: docs/syntax.md (hunk 1 above)
Comment (nadavwix) [reply docs/syntax.md:71]: look at the use cases, if you subscribed you want to unsubscribe

File: docs/syntax.md (hunk 2 above)
Comment (nadavwix) [reply docs/syntax.md:144]: no, not realy

File: docs/syntax.md (hunk 3 above)
Comment (nadavwix) [reply docs/syntax.md:131]: i'd have a maxConsecutive... as a third parameter to afterDomUpdate. and a default of 3, -1 (is infinite)
so by default it's limited

File: docs/syntax.md (hunk 1 above)
Comment (omryn) [reply docs/syntax.md:71]: Yes, the **returned** ("undo") function should be executed, not the action (where you'd subscribe)

**Comment by omryn** [comment]
Implemented in omry/href

**Pull request closed without merging**